src/Orleans.Providers.Legacy.Core/Configuration/BootstrapProviderConfiguration.cs

```diff
@@ -0,0 +1,41 @@
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using Orleans.Providers;
+
+namespace Orleans.Runtime.Configuration
+{
+    public static class BootstrapProviderConfiguration
+    {
+        public const string BOOTSTRAP_PROVIDER_CATEGORY_NAME = "Bootstrap";
+
+        /// <summary>
+        /// Registers a given type of <typeparamref name="T"/> where <typeparamref name="T"/> is bootstrap provider
+        /// </summary>
+        /// <typeparam name="T">Non-abstract type which implements <see cref="IBootstrapProvider"/> interface</typeparam>
+        /// <param name="providerName">Name of the bootstrap provider</param>
+        /// <param name="properties">Properties that will be passed to bootstrap provider upon initialization</param>
+        public static void RegisterBootstrapProvider<T>(this GlobalConfiguration config, string providerName, IDictionary<string, string> properties = null) where T : IBootstrapProvider
```

Comment:
Why do you have this in the legacy *Core* library, since it's only usable in a silo?

Reply:
As a general approach every thing I moved from abstractions or core was moved to core.  So this is in core because original code was in core or abstractions.  Since bootstrap providers are silo only, I can probably move this to runtime, but not all providers are silo only, so some core logic will still be needed :/

Reply:
I don't know... Do we really need to be that fine-grained? After all, this is just a legacy package that we hope users do not have to use. For the ones who do, just having a single legacy package, even if it brings up a few more dependencies than what they need, it might be OK. Not sure we need to optimize for that. On the other side, having an explosion of Legacy packages that we don't intend many will use will just be noise.
I *think* I would even prefer that we have a single or *very* few legacy packages, and not even 1 for just the `Providers` package, such as `Microsoft.Orleans.Legacy`, where we put this, we put the legacy configuration abstractions, legacy logger, etc

/cc @attilah @xiazen @sergeybykov

Reply:
I agree. Or we have two, one for silo, one for client. 
I suggested it because, most legacy features aren't removed, they just live in our new core code in a better form. In other words, their core logic is till living in orleans core code. What we are maintaining in these legacy packages are really just legacy ways to configuring them. Hence we really just need one package for the legacy configuring methods.

If we have a package called `Legacy.Core`  for example, it sounds like it contains some core business logic. 

but Legacy.Logger package is a bit different, because it actually have lots of business logic in it. It contains legacy features which we want to discard in the future, and it also contains feature like bulk we wanted to support continuely. I'm not against to put it into the big legacy package, just mention my thoughts on this.

Reply:
I'm ok with legacy packages not being per feature, but due to the pre-existing dependencies we need at least a core and a runtime version.
What do you think of merging ..Providers.Legacy.Core, Providers.Legacy.Runtime and Legacy.Logger into Orleans.Legacy.Core and Orleans.Legacy.Runtime?

Reply:
@jason-bragg I think it should be fine to have 2, although not sure about which dependencies differences there are that concern you.

BTW, I believe it might be more clear if `Legacy` is the suffix, and not in the middle of the name, as it is extending Core with legacy features.
This way they packages would be:
* `Microsoft.Orleans.Core.Legacy`
* `Microsoft.Orleans.Runtime.Legacy`

Still, it's just a little misleading, as these packages will have a little more dependencies than just Core or Runtime (for example Providers), but I think I can live with that.

@xiazen regarding the Logger functionality, I think we just have bulking there because of legacy reasons, it will not be our recommendation to tell users to use the legacy logging abstraction just to have bulking, and instead do bulking on `ILogger` natively. So I would not consider Logger to be a special case, and instead would bundle it with the rest of the legacy stuff that we intend to deprecate and eventually stop supporting altogether.

Reply:
what do you mean by bulking on ILogger natively? They support bulking feature natively?

Reply:
not built-in (that I know of), what I mean is to have a similar functionality on top of ILogger, without having to go through our `ILogConsumer` abstraction.
If they are using a 3rd party logger to write to files or external stores, they can benefit from the bulking in those logging frameworks. Regardless of current availability, the only reason we still have that is because of legacy... otherwise it shouldn't be in a legacy package (but I'll push back if we want to make bulking 1st class citizen of Orleans, it's not our concern, it's the concern of logging frameworks)

Reply:
but our bulking is built on top of ILogger, not ILogConsumer. Even it is not Orleans core related, but because of the historical reason, we are supporting it as a feature. Either you call it legacy or not, I'm just saying it is not a deprecated feature. I'm not against to put it into the Legacy.Runtime, Legacy.Core package. I was just saying , to me , those Legacy.Runtime, Legacy.Core contains legacy configuration methods which are to be deprecated in the future, but bulking is not. So when we remove all the deprecated pieces, bulking will still be in the legacy package. At that time, we might discuss different options to deal with it, either move it to Orleans Contrib, or keep it in Orleans tree but rename the package name to be Logging.Bulking or something. 
Or we can keep logging legacy features in a separate package now.